### PR TITLE
Allow to customize the exit code send in cli context

### PR DESCRIPTION
**Pull request by lode**
Fixes https://github.com/filp/whoops/issues/688

### CODE REVIEW

File: src/Whoops/Run.php [hunk 1]
```diff
@@ -288,6 +293,31 @@ public function sendHttpCode($code = null)
...
+     */
+    public function sendExitCode($code = null)
+    {
+        if (func_num_args() == 0) {
+            return $this->sendExitCode;
+        }
+
+        if ($code < 0 || 255 <= $code) {
+            throw new InvalidArgumentException(
+                "Invalid status code '$code', must be between 0 and 254"
```
Comment (denis-sokolov) [inline src/Whoops/Run.php:314]: To match the conditional above:

```suggestion
                "Invalid status code '$code', must be between 1 and 255"
```

File: tests/Whoops/RunTest.php [hunk 2]
```diff
@@ -503,4 +503,33 @@ public function testSendHttpCodeWrongCode()
...
+    }
+
+    /**
+     * @covers Whoops\Run::sendExitCode
+     */
+    public function testSendExitCodeWrongCode()
+    {
+        $this->expectExceptionOfType('InvalidArgumentException');
+
+        $this->getRunInstance()->sendExitCode(255);
```
Comment (denis-sokolov) [inline tests/Whoops/RunTest.php:533]: How does this test pass? The condition allows 255 with less-or-equal operator.

**Review by denis-sokolov — CHANGES_REQUESTED** [review]
Superbly clean, thank you!

Some clarification with the allowed range is needed, could you take a look and clarify what you meant? Thanks.

File: tests/Whoops/RunTest.php (hunk 2 above)
Comment (lode) [reply tests/Whoops/RunTest.php:533]: It works for me. The conditional is about when to throw, not when to allow.

File: src/Whoops/Run.php (hunk 1 above)
Comment (denis-sokolov) [reply src/Whoops/Run.php:314]: I’ve misread the conditional, I’ve read it as `!(0 < $code && $code <= 255)`. Yeah, your reasoning is stellar, thanks for the patient answer!

File: tests/Whoops/RunTest.php (hunk 2 above)
Comment (denis-sokolov) [reply tests/Whoops/RunTest.php:533]: Indeed, I’ve misread it. Thanks!

**Comment by lode** [comment]
Thanks a lot for the quick working along!

**Pull request merged**